core/src/main/java/org/mule/runtime/core/internal/processor/strategy/StreamPerThreadSink.java

```diff
@@ -129,9 +89,8 @@ public void dispose() {
         LOGGER.warn("TX Subscribers of ProcessingStrategy for flow '{}' not completed before thread interruption",
                     flowConstruct.getName());
       }
-      sinks.invalidateAll();
-      sinksNestedTx.invalidateAll();
-    } else if (disponsableSinks.get() != 0) {
+      sinkProvider.invalidateAll();
```

Comment:
maybe de invalidateAll should not be public and has to be in the dispose (the sinkProvider should be Disposable) Can the disposableSinks counter be internal to the CacheSinkProvider?